### ISSUE TITLE
[BUG] Got response of unexpected request

### ISSUE THREAD

**Issue by 304NotModified** [microsoft/playwright-dotnet#2101]
**Context:**
- Playwright Version: 1.20.1
- Operating System: Windows
- .NET version: .NET 5
- Browser: Chromium
- Extra: this is an open id connect flow

**Code Snippet**

```cs
var response = await page.GotoAsync("https://mysite/overzicht", new PageGotoOptions { WaitUntil = WaitUntilState.NetworkIdle });
var responseUrl = response!.Url;
var responseHeaders = await response.AllHeadersAsync();
```

**Describe the bug**
Going to this url will do 4 requests:

![image](https://user-images.githubusercontent.com/5808377/163566282-ac9ddeee-d30e-447b-8868-0c6885a0fcd2.png)

**Expected**
- Get the URL and headers of the **last** request. [The docs](https://playwright.dev/dotnet/docs/api/class-page#page-goto) also say 
   > Returns the main resource response. In case of multiple redirects, the navigation will resolve with the response of the **last** redirect.

**Current**
- The page is on "overzicht" (the last request) - that's good!
- The _responseUrl_  and the _responseHeaders_  are of the 2nd request - "authorize".

**Comment by campersau** [comment]
For me this looks correct as the "authorize" request returned a 200 OK response which isn't a redirect. The next request to "signin-oidc" was maybe automatically triggered by javascript or something else and thus **looked like** a redirect but in HTTP terms it isn't one.

**Comment by 304NotModified** [comment]
I can see that. But maybe the difference is also WaitUntilState.NetworkIdle. When I am at the 
`var responseUrl = response!.Url;` with a breakpoint, the page had done already the 4 requests. 

Perhaps we should change the docs to

> Returns the main resource response. In case of multiple redirects, the navigation will resolve with the response of the **first** 2xx response.

**Comment by campersau** [comment]
Yes, I can see how the current documentation can be misleading. But I would phrase it a little differently:

> Returns the main resource response. In case of multiple redirects, the navigation will resolve with the first non-redirect response.

Because this includes all other non-redirecting status codes as well.

**Comment by 304NotModified** [comment]
Besides the doc change - how could I get the response of the latest?  For now, I only see the option by listing to the [Page.Request](https://playwright.dev/dotnet/docs/api/class-page#page-event-request) event?
